### ISSUE TITLE
TempoMonolithic: `affinity` and `tolerations` are not respected by the operator

### ISSUE THREAD

**Issue by skuethe** [grafana/tempo-operator#867]
We are using **tempo-operator** `v0.9.0` and having issues with the new feature added to define `affinity` and `tolerations`.
We can add them to the CR instance, but the operator does not respect the values when creating the statefulset instance.

Output from the applied CR instance named `cluster`:

```yaml
$ k get tempomonolithics.tempo.grafana.com cluster -o yaml | yq '.spec.affinity'
nodeAffinity:
  requiredDuringSchedulingIgnoredDuringExecution:
    nodeSelectorTerms:
      - matchExpressions:
          - key: node-role.kubernetes.io/observability
            operator: Exists

$ k get tempomonolithics.tempo.grafana.com cluster -o yaml | yq '.spec.tolerations'
- effect: NoSchedule
  key: dedicated
  operator: Equal
  value: observability
```

And the statefulset created from the operator:

```yaml
$ k get sts tempo-cluster -o yaml | yq '.spec.template.spec.affinity'
podAntiAffinity:
  preferredDuringSchedulingIgnoredDuringExecution:
    - podAffinityTerm:
        labelSelector:
          matchLabels:
            app.kubernetes.io/component: tempo
            app.kubernetes.io/instance: cluster
            app.kubernetes.io/managed-by: tempo-operator
            app.kubernetes.io/name: tempo-monolithic
        topologyKey: kubernetes.io/hostname
      weight: 100
    - podAffinityTerm:
        labelSelector:
          matchLabels:
            app.kubernetes.io/component: tempo
            app.kubernetes.io/instance: cluster
            app.kubernetes.io/managed-by: tempo-operator
            app.kubernetes.io/name: tempo-monolithic
        topologyKey: failure-domain.beta.kubernetes.io/zone
      weight: 75
$ k get sts tempo-cluster -o yaml | yq '.spec.template.spec.tolerations'
null

```

And just for confirmation, CRD property parsing:

```yaml
$ k get crd tempomonolithics.tempo.grafana.com -o yaml | yq '.spec.versions[].schema.openAPIV3Schema.properties.spec.properties | keys'
- affinity
- extraConfig
- ingestion
- jaegerui
- management
- nodeSelector
- observability
- resources
- storage
- tolerations
```

**Comment by skuethe** [comment]
For completeness:

```bash
$ k exec -it tempo-cluster-0 -- ./tempo --version
tempo, version 2.3.1 (branch: HEAD, revision: 9398fd620)
  build user:       
  build date:       
  go version:       go1.21.3
  platform:         linux/amd64
  tags:             unknown
```

**Comment by andreasgerstmayr** [comment]
@skuethe thanks for the bug report! This issue will be fixed in #880.
